### PR TITLE
CC 2019 FEBVRE

### PR DESCRIPTION
**Pull request by jpcosal**

### CODE REVIEW

File: Exam/src/exercice1/NaturalIntegers.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+public class NaturalIntegers {
+
+	public static int IntegersCubeSum(int nIntegers) {
+		int cube = 0;
+		for (int i = 1;i!=nIntegers+1;i++) {
+			cube += i*i*i;
+		}
+		return cube;
+	}
```
Comment (Grimmss83) [inline Exam/src/exercice1/NaturalIntegers.java]: Que se passe t-il si j'envoie 0 ?

File: Exam/src/exercice1/NaturalIntegersTest.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+	@Test
+	void cubeOfSumFirstNaturalInt() {
+		assertEquals(166375, NaturalIntegers.IntergersSumCube(10));
+	}
+
+	@Test
+	void differenceBetweenCubeOfSumAndSumOfCube() {
+		assertEquals(163350, (NaturalIntegers.IntergersSumCube(10)-NaturalIntegers.IntegersCubeSum(10)));
+	}
+}
```
Comment (Grimmss83) [inline Exam/src/exercice1/NaturalIntegersTest.java]: Test pour 100 ?

File: Exam/src/exercice2/EventItem.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+	String newDate;
+	
+	public EventItem() {
+		time = new Date(2000, 11, 21);
+		formattedDate = new SimpleDateFormat("YY/MM/DD - hh:mm");
+		newDate = formattedDate.format(time);
+	}
+	
+	public String toString() {
+		return newDate;
```
Comment (Grimmss83) [inline Exam/src/exercice2/EventItem.java]: Tu retournes seulement la date ? Et Source/Type etc...

File: Exam/src/exercice2/EventItemTest.java [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+import org.junit.jupiter.api.Test;
+
+class EventItemTest {
+	EventItem item = new EventItem();
+	
+	@Test
+	void test() {
+		assertEquals("1991", item.toString());
+	}
+
```
Comment (Grimmss83) [inline Exam/src/exercice2/EventItemTest.java]: Test non concluant je suppose

File: Exam/src/exercice4/Article.java [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+	public void setNom(String name) {
+		this.name = name;
+	}
+	public void setPriceTaxeFree(double priceTaxeFree) {
+		this.priceTaxeFree = priceTaxeFree;
+	}
+	public void setRef(int ref) {
+		this.ref = ref;
+	}
+}
```
Comment (Grimmss83) [inline Exam/src/exercice4/Article.java]: Pourquoi ne pas mettre public au lieu de mettre des getter/setter ?

File: Exam/src/exercice1/NaturalIntegers.java (hunk 1 above)
Comment (jpcosal) [reply Exam/src/exercice1/NaturalIntegers.java]: I think you know :D

File: Exam/src/exercice2/EventItem.java [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+public class EventItem {
+	SimpleDateFormat formattedDate;
+	Date time;
+	Object Source;
+	String type;
+	String newDate;
```
Comment (jpcosal) [inline Exam/src/exercice2/EventItem.java]: This is **computed**!
Don't store that in an attribute

File: Exam/src/exercice2/EventItem.java [hunk 7]
```diff
@@ -0,0 +1,23 @@
+package exercice2;
+
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+public class EventItem {
+	SimpleDateFormat formattedDate;
```
Comment (jpcosal) [inline Exam/src/exercice2/EventItem.java]: You need this only to compute something. Why store that as an attriubte??

File: Exam/src/exercice2/EventItem.java (hunk 3 above)
Comment (jpcosal) [inline Exam/src/exercice2/EventItem.java]: This is not exactly what was expected

File: Exam/src/exercice3/BabyToy.java [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+	
+	public void addATriangle(Triangle theTriangle) {
+		listOfTriangle.add(theTriangle);
+	}
+	
+	public void addASquare(Square theSquare) {
+		listOfSquare.add(theSquare);
+	}
+	
+	public double getAreaOfTheToy() {
```
Comment (jpcosal) [inline Exam/src/exercice3/BabyToy.java]: Works but... Imagine you have to maintain this code and support like 1000 more basic shapes...

File: Exam/src/exercice3/BabyToyTest.java [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+class BabyToyTest {
+	BabyToy babyToy;
+	Circle aCircle;
+	Triangle aTriangle;
+	Square aSquare;
+	double expectedArea;
+	
+	
+	@BeforeEach
+	void setUp() throws Exception {
```
Comment (jpcosal) [inline Exam/src/exercice3/BabyToyTest.java]: Hmmm... a fixture for a single test case??

File: Exam/src/exercice4/CashierUtils.java [hunk 10]
```diff
@@ -0,0 +1,33 @@
+package exercice4;
+
+import java.util.Date;
+import java.util.List;
+
+public class CashierUtils {
+	public double TVA = 1.20;
+	
+	public static String getMensualReport(List<Sale> sales,int MM, int YY) {
```
Comment (jpcosal) [inline Exam/src/exercice4/CashierUtils.java]: What exactly represents input `sales`, and **who** should provide this information?

**Pull request closed without merging**